src/lua/compat.lua

```diff
@@ -0,0 +1,236 @@
+-- compat.lua -- internal module intended to solve compatibility problems in
+-- different parts of Tarantool. Introduced in gh-7000, see also gh-6912.
+
+local options_format = {
+    old     = "boolean",
+    new     = "boolean",
+    default = "boolean",
+    frozen  = "boolean",
+    brief   = "string",
+    doc     = "string"
+}
+
+-- Contains static options descriptions in format specified by `options_format`.
+local options = { }
+
+-- Contains postaction functions for each unfrozen option from `options`.
+local postaction = { }
+
+-- Contains dinamic values: current state of an option and is it selected.
+local cfg = { }
+
+local help = [[
+
+This is Tarantool compatibility module.
+To get help, see the Tarantool manual at https://tarantool.io/en/doc/.
+Available commands:
+
+    candidates()                    -- list all unselected options
+    get_setup()                     -- get lua command that sets up different compat with same options as current
+    help()                          -- show this help
+    reset()                         -- set all options to default
+    add_options{{name, {}, action}} -- add new options by providing name, option table and postaction for each one
```

Comment:
This was the closest to param example that I could fit into a help msg. It is `option table` (not optional), as you can see in tests and usage examples. My point was to show expected input type, but I can see how it can be misunderstood. I am open to any suggestions on how to make the message clearer.

Reply:
Changed to `add_options()`.